#511 Add accessible slash command arguments

symphony-soufiane opened the pull request:
### Description
Closes #238 
In this commit, we made slash command support string/cashtag/hashtag/mention arguments. We introduced slashCommandPattern class to represent the pattern to match user input.
When user input matches the pattern provided in the slash command, we add arguments values in the commandContext in order to be used in the command callback logic.
When a command required bot mention, we check bot user id instead of the display name.
The command pattern is tokenised and user input as well. The matching consists of comparing the tokens values.

### Checklist
- [x] Referenced an issue in the PR title or description
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Docstrings added or updated
- [x] Updated the documentation in [docsrc folder](../tree/main/docsrc)

symphony-soufiane committed:
#511 Add accessible slash command arguments
In this commit, we made slash command support string/cashtag/hashtag/mention arguments. We introduced slashCommandPattern class to represent the pattern to match user input.
When user input matches the pattern provided in the slash command, we add arguments values in the commandContext in order to be used in the command callback logic.
When a command required bot mention, we check bot user id instead of the display name.
The command pattern is tokenized and user input as well. The matching consists of comparing the tokens values.

symphony-soufiane commented on tests/core/activity/parsing/input_tokenizer_test.py:
```diff
@@ -0,0 +1,146 @@
+    assert under_test.tokens == ["lorem", "jane-doe"]
+
+
+# TODO: fix this test
```
Will be fixed in next commit

symphony-soufiane commented on tests/core/activity/parsing/slash_command_pattern_test.py:
```diff
@@ -0,0 +1,286 @@
+        SlashCommandPattern("{$$}")
+
+
+# TODO: fix this test
```
Will be fixed in next commit

symphony-youri commented on symphony/bdk/core/activity/parsing/arguments.py:
```diff
@@ -0,0 +1,90 @@
+        return None
+
+    @property
+    def arguments(self):
```
do we want to give direct access to the underlying map?

symphony-soufiane replied:
You're right, the other getters (get_string, get_mention..) should be enough. This will be removed in next commit.

symphony-soufiane committed:
Make bot user id loaded in a lazy way during runtime, using a lambda function.
Hide arguments map from Arguments object as we do not want to expose it. Added unit tests for Arguments class.

symphony-soufiane committed:
Support words inside tags in InputTokenizer

symphony-soufiane committed:
Merge branch 'main' into support-slash-arguments
